tests/core/polynomials_test.py

```diff
@@ -0,0 +1,329 @@
+        numpy_module_mock = set_up_mocks()
+
+        # Mocked return values of called functions
+        numpy_module_mock.power.side_effect = [array([1]), array([2]), array([4])]
+        numpy_module_mock.zeros_like.return_value = array([0])
+
+        with patch.dict(
+            "sys.modules",
+            {
+                "numpy": numpy_module_mock,
+            },
+        ):
+            from napytau.core.polynomials import (
+                evaluate_polynomial_at_measuring_distances,
+            )
+
+            distances: ndarray = array([2])
+            coefficients: ndarray = array([1, 2])
+            # Polynomial: f(x) = 1 + 2x
+            # At x = 2: 1 + 2(2) = 5
+            expected_result: ndarray = array([5])
+            testing.assert_array_equal(
+                evaluate_polynomial_at_measuring_distances(distances, coefficients),
+                expected_result,
+            )
+
+    @staticmethod
+    def test_CanEvaluateASingleValuePolynomialAtMeasuringDistances():
+        """Can evaluate a single value polynomial at measuring distances."""
+        numpy_module_mock = set_up_mocks()
+
+        # Mocked return values of called functions
+        numpy_module_mock.power.side_effect = [
+            array([1, 1, 1]),
+            array([1, 2, 3]),
+            array([1, 4, 9]),
+        ]
+        numpy_module_mock.zeros_like.return_value = array([0, 0, 0])
+
+        with patch.dict(
+            "sys.modules",
+            {
+                "numpy": numpy_module_mock,
+            },
+        ):
+            from napytau.core.polynomials import (
+                evaluate_polynomial_at_measuring_distances,
+            )
+
+            distances: ndarray = array([1, 2, 3])
+            coefficients: ndarray = array([5])
+            # Constant polynomial: f(x) = 5
+            # All values should be 5
+            expected_result: ndarray = array([5, 5, 5])
+            testing.assert_array_equal(
+                evaluate_polynomial_at_measuring_distances(distances, coefficients),
+                expected_result,
+            )
+
+    @staticmethod
+    def test_CanEvaluateAnEmptyPolynomialAtMeasuringDistances():
+        """Can evaluate an empty polynomial at measuring distances."""
+        numpy_module_mock = set_up_mocks()
+
+        # Mocked return values of called functions
+        numpy_module_mock.zeros_like.return_value = array([0, 0])
+
+        with patch.dict(
+            "sys.modules",
+            {
+                "numpy": numpy_module_mock,
+            },
+        ):
+            from napytau.core.polynomials import (
+                evaluate_polynomial_at_measuring_distances,
+            )
+
+            distances: ndarray = array([1, 2])
+            coefficients: ndarray = array([])
+            # With an empty coefficients array, the result should be an array of zeros
+            # with the same length as 'distances'
```

Comment:
One could argue, that an empty polynomial should lead to an error, as it is not a valid function. A polynomial with only one coefficient that is 0 would be a valid function:

```
f(x) = 
vs
f(x) = 0
```

Reply:
=> i would expect the function to throw an error in this case, consider adding a `CoreError` for this.

Reply:
I agree, I think I mixed up 0 parameters and a function of degree 0 in my head.